extensions/mapstore/src/main/java/com/hazelcast/mapstore/GenericMapStore.java

```diff
@@ -209,15 +209,18 @@ private String deriveMappingType() {
         if (properties.mappingType != null) {
             return properties.mappingType;
         } else {
-            DataLinkFactory<?> factory = nodeEngine()
+            try (DataLink dataLink = nodeEngine()
                     .getDataLinkService()
-                    .getDataLinkFactory(properties.dataLinkRef);
+                    .getDataLink(properties.dataLinkRef)) {
 
-            if (factory instanceof JdbcDataLinkFactory) {
-                return "JDBC";
-            } else {
-                throw new HazelcastException("Unknown DataLinkFactory class " + factory.getClass()
-                        + ". Set the mapping type using '" + MAPPING_TYPE_PROPERTY + "' property");
+                if (dataLink instanceof JdbcDataLink) {
+                    return "JDBC";
```

Comment:
maybe method `mappingTypeForLink()` in `DataLink` (or something like that); otherwise this method will contain a lot of dependencies also to other modules